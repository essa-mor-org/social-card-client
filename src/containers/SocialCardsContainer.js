import React, { PureComponent } from 'react';
import { connect } from 'react-redux';

import { fetchPosts } from '../actions/Posts';
import { postsSelector } from '../selectors/Posts';
import SocialCardContainer from './SocialCardContainer';

export class SocialCardsContainer extends PureComponent {
	componentDidMount() {
		this.props.fetchPosts();
	}

	render() {
		const { posts } = this.props;
		const postItems = posts.valueSeq().map(post => (<SocialCardContainer
			key={post.get('id')}
			id={post.get('id')}
			image={post.get('image')}
			profileText={post.get('profileText')}
			time={post.get('time')}
			text={post.get('text')}
			postImage={post.get('postImage')}
			likes={post.get('likes')}
			shares={post.get('shares')}
			profileImage={post.get('profileImage')} />));
		return (<div>{postItems}</div>);
	}
}

const mapStateToProps = (state) => ({
	posts: postsSelector(state)
});

export default connect(mapStateToProps, { fetchPosts })(SocialCardsContainer);