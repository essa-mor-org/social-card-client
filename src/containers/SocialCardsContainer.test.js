import { fromJS } from 'immutable';
import React from 'react';
import MockProvider, { getMockStore } from 'redux-mock-provider';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { SocialCardsContainer } from './SocialCardsContainer';

const posts = fromJS({
	'1': {
		'id': 1,
		'image': 'https://scontent.ftlv4-1.fna.fbcdn.net/v/t1.0-1/p80x80/19554009_10154562736535933_8492592550808364419_n.jpg?_nc_cat=0&oh=e2dcab39740d500b15ab15c33715c1cb&oe=5BEA19A5',
		'profileText': 'Vevo',
		'time': '3 hrs',
		'text': 'Calling all Guns N\' Roses fans! Here\'s your chance to win the Appetite For Destruction Super Deluxe Box Set‼️ Enter now for your chance to win!',
		'postImage': 'https://scontent.ftlv4-1.fna.fbcdn.net/v/t1.0-9/36907526_10155523573880933_4550184912837148672_n.jpg?_nc_cat=0&oh=907c9bbce64ffe2808a821fb673f140d&oe=5BA005E7',
		'likes': 157,
		'shares': 3,
		'profileImage': 'https://scontent.ftlv4-1.fna.fbcdn.net/v/t1.0-1/p48x48/14572897_10209280775468762_3731006062202010506_n.jpg?_nc_cat=0&oh=41eba23bde8e0c43e709a6a9f9867c55&oe=5BDD6AD2',
		'profileName': "Essa Mor",
		"postComments": []
	}
});

const store = getMockStore({
	key: 'entities',
	state: {}
});

store.getState = () => fromJS({
	'entities':
	{
		'posts':
			{ 'byId': { '1': { 'profileImage': 'https://scontent.ftlv4-1.fna.fbcdn.net/v/t1.0-1/p48x48/14572897_10209280775468762_3731006062202010506_n.jpg?_nc_cat=0&oh=41eba23bde8e0c43e709a6a9f9867c55&oe=5BDD6AD2' } } }
	}
});

describe('SocialCardsContainer test', () => {
	it('renders correctly', () => {
		const tree = renderer
			.create(<MockProvider store={store}><SocialCardsContainer posts={posts}
				fetchPosts={jest.fn()} likePost={jest.fn()} sharePost={jest.fn()} /></MockProvider>)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('call fetchPosts', () => {
		const callBack = jest.fn();
		mount(<MockProvider store={store}><SocialCardsContainer posts={posts}
			fetchPosts={callBack} likePost={jest.fn()} sharePost={jest.fn()} /></MockProvider>);
		expect(callBack).toHaveBeenCalledTimes(1);
	});
});
