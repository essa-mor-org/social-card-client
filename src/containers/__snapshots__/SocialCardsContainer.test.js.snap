// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialCardsContainer test renders correctly 1`] = `
<div>
  <div
    className="Social_card"
  >
    <div
      className="Social_header"
    >
      <div
        className="Social_header_column1"
      >
        <a
          className="Social_header_image_link"
          href="#"
        >
          <img
            alt=""
            className="Social_header_image"
            src="https://scontent.ftlv4-1.fna.fbcdn.net/v/t1.0-1/p80x80/19554009_10154562736535933_8492592550808364419_n.jpg?_nc_cat=0&oh=e2dcab39740d500b15ab15c33715c1cb&oe=5BEA19A5"
          />
        </a>
        <a
          className="Social_header_name"
          href="#"
        >
          Vevo
        </a>
        <a
          className="Social_header_time"
          href="#"
        >
          3 hrs
        </a>
        <span
          className="Social_header_saperator"
        >
          .
        </span>
        <a
          className="Social_header_world"
          href="#"
        />
      </div>
      <a
        className="Social_header_toggle"
        href="#"
      />
    </div>
    <div
      className="Social_post_text"
    >
      Calling all Guns N' Roses fans! Here's your chance to win the Appetite For Destruction Super Deluxe Box Set‼️ Enter now for your chance to win!
    </div>
    <a
      href="#"
    >
      <img
        alt=""
        className="Social_post_image"
        src="https://scontent.ftlv4-1.fna.fbcdn.net/v/t1.0-9/36907526_10155523573880933_4550184912837148672_n.jpg?_nc_cat=0&oh=907c9bbce64ffe2808a821fb673f140d&oe=5BA005E7"
      />
    </a>
    <div
      className="Social_reaction"
    >
      <a
        className="Social_reaction_fixed"
        href="#"
      >
        <span
          className="Social_reaction_like"
        />
      </a>
      <span
        className="Social_reaction_strech"
      >
        <a
          className="Social_reaction_number"
          href="#"
        >
          157
        </a>
      </span>
      <a
        className="Social_reaction_number Social_reaction_fixed"
        href="#"
      >
        2
         Comments
      </a>
      <a
        className="Social_reaction_number Social_reaction_fixed"
        href="#"
      >
        3
         Shares
      </a>
    </div>
    <div
      className="Social_react"
    >
      <a
        className="Reaction_like Reaction_button"
        href="#"
        onClick={[Function]}
      >
        Like
      </a>
      <a
        className="Reaction_comment Reaction_button"
        href="#"
        onClick={[Function]}
      >
        Comment
      </a>
      <a
        className="Reaction_share Reaction_button"
        href="#"
        onClick={[Function]}
      >
        Share
      </a>
    </div>
    <div
      className="Social_comments_input_panel"
    >
      <img
        alt=""
        className="Social_comments_image"
        src="https://scontent.ftlv4-1.fna.fbcdn.net/v/t1.0-1/p48x48/14572897_10209280775468762_3731006062202010506_n.jpg?_nc_cat=0&oh=41eba23bde8e0c43e709a6a9f9867c55&oe=5BDD6AD2"
      />
      <input
        className="Social_comments_input"
        onKeyPress={[Function]}
        placeholder="Write a comment..."
        type="text"
      />
    </div>
    <div
      className="Comment_panel"
    >
      <img
        alt=""
        className="Social_comments_image"
        src="https://scontent.ftlv4-1.fna.fbcdn.net/v/t1.0-1/c0.0.48.48/p48x48/30571266_10211755395060285_621329575565066240_n.jpg?_nc_cat=0&oh=755d9aa650987a5b1519cdbaeeedcd8c&oe=5BEA6497"
      />
      <div
        className="Comment_comment_panel"
      >
        <a
          className="Comment_profile"
          href="#"
        >
          Tyler Schmidt
        </a>
        <span
          className="Comment_comment"
        >
          Written by Lady Gaga
        </span>
      </div>
    </div>
    <div
      className="Comment_panel"
    >
      <img
        alt=""
        className="Social_comments_image"
        src="https://scontent.ftlv4-1.fna.fbcdn.net/v/t1.0-1/c0.0.48.48/p48x48/30571266_10211755395060285_621329575565066240_n.jpg?_nc_cat=0&oh=755d9aa650987a5b1519cdbaeeedcd8c&oe=5BEA6497"
      />
      <div
        className="Comment_comment_panel"
      >
        <a
          className="Comment_profile"
          href="#"
        >
          Tyler Schmidt
        </a>
        <span
          className="Comment_comment"
        >
          Man they confiscated my pages Have a great day
        </span>
      </div>
    </div>
  </div>
</div>
`;
